platform/engineering/design-docs/2022-01-12-Elixir.md

```diff
@@ -0,0 +1,100 @@
+# Elixir SignInService Design Doc
+
+**Author(s):** Trevor Bosaw, Joe Niquette, VSP-Identity  
+**Last Updated:** January 12, 2022  
+**Status:** **Draft**   
+**Approvers:** Michael Fleet \[ \]
+
+**Keywords:** Agile mindset, rapid innovation, zero trust architecture, microservice, Elixir, Erlang, Phoenix Framework 
+
+## Overview
+
+### Objective
+  Utilize the Elixir programming language to implement an authentication and authorization sign-in service with heavy concurrency and high availability needs
+Foster a community within the organization to share and expand Elixir knowledge, in order to encourage usage towards additional relevant implementations
+Intended audience is engineers within VSP who likely will interact with services that are implemented with the Elixir language. Otherwise there should be little to no requirement of other VA employees to understand Elixir as it's being used by VA.gov.
+ 
+
+### Background
+  The sign in service is intended to be a new solution for what will eventually be the authentication and authorization flow for all VA.gov users. This service will have clear concurrency and performance needs that do not make it a good candidate for an implementation inside the vets-api monolith, but rather encourage a separation as its own separate service. Additionally, the organization itself has encouraged a microservice approach for new features, and this feature is particularly suited for this new approach. Other reasons for separating this service from the vets-api monolith are to allow for simplified testing, and clear service boundaries. Finally, scalability is a concern for the new sign in service, as authentication and authorization mechanisms in vets-api today represent a bottleneck in the application. Therefore, it is imperative that this service is implemented with performance and concurrency considerations. Otherwise, it is likely that we would necessitate increases in our overall infrastructure instance count, rather than reductions 
```

Comment:
Do we want to mention that this is going to scale beyond vets-api (since we're already handling MHV's sign in through vets-api) as another reason for needing excellent scalability. We could also elaborate that this is the best example of clear service boundaries - we don't want vets-api handling authentication for a sign in request that has nothing to do with vets-api.

Reply:
Yeah I think that's important context. *I* know it, but all readers of this doc may not.

Reply:
I added some language in for this, if you could help add more I'd appreciate it. Im going to merge this in a few hours, future changes should open another PR